prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -2466,19 +2287,31 @@ class Hl7Serializer(
             return ""
         }
     }
-}
 
-/**
- * Trim and truncate the string to the [maxLength] preserving as much of the non-whitespace as possible
- */
-fun String.trimAndTruncate(maxLength: Int?): String {
-    val startTrimmed = this.trimStart()
-    val truncated = if (maxLength != null && startTrimmed.length > maxLength) {
-        startTrimmed.take(maxLength)
-    } else {
-        startTrimmed
+    /**
+     * Delegate call to universal pipeline truncation logic
+     */
+    internal fun trimAndTruncateValue(
```

Comment:
Added these functions to parse the configs from the Covid pipeline and then delegate to the shared code.